### ISSUE TITLE
AttributeError: 'tuple' object has no attribute 'items'

### ISSUE THREAD

**Issue by jacebrowning** [tomchristie/flask-api#61]
Related to #58,  I am now hitting a spot where `blueprint_handlers` and `app_handlers` are both empty tuples with Flask 0.11 and 0.11.1, so the fix in #59 actually causes a new issue:

```
Traceback (most recent call last):
  File ".../env/lib/python3.5/site-packages/flask/app.py", line 2000, in __call__
    return self.wsgi_app(environ, start_response)
  File ".../env/lib/python3.5/site-packages/flask/app.py", line 1991, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File ".../env/lib/python3.5/site-packages/flask/app.py", line 1567, in handle_exception
    reraise(exc_type, exc_value, tb)
  File ".../env/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File ".../env/lib/python3.5/site-packages/flask/app.py", line 1988, in wsgi_app
    response = self.full_dispatch_request()
  File ".../env/lib/python3.5/site-packages/flask/app.py", line 1641, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File ".../env/lib/python3.5/site-packages/flask_api/app.py", line 97, in handle_user_exception
    for typecheck, handler in chain(blueprint_handlers.items(), app_handlers.items()):
AttributeError: 'tuple' object has no attribute 'items'
```

**Comment by jacebrowning** [comment]
Looking at [these lines](https://github.com/tomchristie/flask-api/blob/ab4252d9cd6e71a74e28b314fd498afd53c15a44/flask_api/app.py#L87-L91), it seems that empty tuples are the default values.

**Comment by dutronlabs** [comment]
Investigating, will get back to you shortly.

**Comment by bogolla** [comment]
Changing line [97](https://github.com/tomchristie/flask-api/blob/ab4252d9cd6e71a74e28b314fd498afd53c15a44/flask_api/app.py#L97) from:
`for typecheck, handler in chain(blueprint_handlers.items(), app_handlers.items()):`
to:
`for typecheck, handler in chain(dict(blueprint_handlers).items(), dict(app_handlers).items()):`
fixes the issue. I'v tried to inspect this and depending on your implementation for a custom exception/error handler, the value for app_handlers or blueprint_handlers will be a tuple. I just typecast that value to a dict and everything else works fine.

**Comment by jacebrowning** [comment]
@bogolla That sounds like a plan! Would you like to submit a PR for this change?

**Comment by bogolla** [comment]
@jacebrowning i have submitted the [PR](https://github.com/tomchristie/flask-api/pull/62)

**Comment by altmind** [comment]
Flask==0.11.1
Flask-API==0.6.7
Also hit this issue - just throw any exception inside controller method. It gets swallowed and instead of original exception we got
```
 File "... site-packages\flask_api\app.py", line 97, in handle_user_exception
    for typecheck, handler in chain(blueprint_handlers.items(), app_handlers.items()):
AttributeError: 'tuple' object has no attribute 'items'
```

Is there something preventing PR #62 from being accepted?

**Comment by jacebrowning** [comment]
@altmind Are you able to test #62 in your environment to confirm the fix?

**Comment by altmind** [comment]
@jacebrowning i'm quite new to python, i dont know if it is the _right_ fix, but it definitely fixed the exception handling for me :)

**Comment by jacebrowning** [comment]
This should now be fixed in `Flask-API == 0.6.8`.
